src/java.base/share/classes/java/util/zip/ZipFile.java

```diff
@@ -1768,18 +1768,17 @@ private void initCEN(int knownTotal) throws IOException {
             // Iterate through the entries in the central directory
             int idx = 0; // Index into the entries array
             int pos = 0;
-            int entryPos = CENHDR;
-            int limit = cen.length;
             manifestNum = 0;
-            while (entryPos <= limit) {
+            while (pos <= cen.length - CENHDR) {
```

Comment:
The interpreter would probably like something like: 
```
int limit = cen.length - CENHDR;
while (pos <= limit) {
```

Reply:
Fixed, look like a good improvement.